modules/apps/commerce/commerce-order-content-web/src/main/java/com/liferay/commerce/order/content/web/internal/importer/type/CSVCommerceOrderImporterTypeImpl.java

```diff
@@ -122,15 +125,35 @@ public List<CommerceOrderImporterItem> getCommerceOrderImporterItems(
 			_commerceChannelLocalService.getCommerceChannelByOrderGroupId(
 				commerceOrder.getGroupId());
 
+		_commerceOrderImporterItemImpls = _getCommerceOrderImporterItemImpls(
+			commerceOrder.getCompanyId(), commerceChannel.getGroupId(),
+			(FileEntry)object);
+
+		int start = 0;
+		int end = _commerceOrderImporterItemImpls.length;
+
+		if (fdsPagination != null) {
+			start = fdsPagination.getStartPosition();
+
+			if (fdsPagination.getEndPosition() < end) {
+				end = fdsPagination.getEndPosition();
+			}
+		}
+
 		return CommerceOrderImporterTypeUtil.getCommerceOrderImporterItems(
 			_commerceContextFactory, commerceOrder,
-			_getCommerceOrderImporterItemImpls(
-				commerceOrder.getCompanyId(), commerceChannel.getGroupId(),
-				(FileEntry)object),
+			Arrays.copyOfRange(_commerceOrderImporterItemImpls, start, end),
 			_commerceOrderItemService, _commerceOrderPriceCalculation,
 			_commerceOrderService, _userLocalService);
 	}
 
+	@Override
+	public int getCommerceOrderImporterItemsCount(Object object)
+		throws Exception {
+
+		return _commerceOrderImporterItemImpls.length;
```

Comment:
This could be null if the method is called before the get....Items.
I would like to add a null check, and call again
_commerceOrderImporterItemImpls = _getCommerceOrderImporterItemImpls(.....);
if null.